src/TaskRunner.ts

```diff
@@ -73,12 +75,16 @@ class TaskRunner {
      */
     addTask<T>(taskName: string, dependencies?: string | string[] | Task<T>, task?: Task<T>): void {
         if (taskName === null || taskName === undefined) {
-            throw new Error("Missing task name");
+            this.reportFatalError(new Error("Missing task name"));
+            return;
```

Comment:
I can see the use-case for listening for fatal errors that result in an execution failure (rather than just rely on `run().catch`), but why on preconditions where it's an API usage error?

As this commit is written, if users don't attach an `onFatalError` callback they will never know if they incorrectly used `addTask` and others.

Reply:
This was mostly to keep it consistent with all the other fatal errors that were not task related, and make it up to the consumer how to handle every error that is thrown. The main use case for this being the CLI type usage, where you require and evaluate a js file. This allows you to nicely handle task runner exceptions even for stuff async stuff. However in that case you will have to wrap stuff in a try catch anyway, so the its not a huge need.

You're right, `onFatalError` should be required to be handled, and `DEFAULT_OPTIONS` should provide a default version that just throws an unhandled exception like before.

Reply:
I'm onboard with having onFatalError be required, with a default version that throws. Thanks, @BuckeyeDude